azavea/district-builder-dtl-pa
# Persist Plan Submissions to the database

## Overview

Creates a PlanSubmission model and uses it to persist contest submissions to the database.

### Checklist


### Notes

- This is slower than I would prefer for an interactive endpoint, but at least the UI already has a nice loading window, and it's not out of line with the rest of the app.

## Testing Instructions

 * Run migrations via `docker-compose exec django bash` and then `./manage.py migrate`
 * Go to the share tab and submit your plan, verify that it works.
 * Play with the parameters and confirm that you _can_ submit without optional fields and _can't_ submit without required ones.
 * Fire up a Django shell with `./manage.py shell_plus` and then confirm that you have a PlanSubmission object with `PlanSubmission.objects.all().count()`. 
 * Get a PlanSubmission object and confirm that `submission.plan.name` is different from the name of whatever plan you submitted.
 * Delete your Plan and confirm that the PlanSubmission object still exists.
